src/axom/primal/operators/winding_number.hpp

```diff
@@ -0,0 +1,391 @@
+{
+/*
+ * \brief Compute the winding number with respect to a line segment
+ *
+ * \param [in] q The query point to test
+ * \param [in] s The line segment
+ * \param [in] edge_tol The tolerance at which a point is on the line
+ *
+ * \return double The generalized winding number
+ */
+template <typename T>
+double winding_number(const Point<T, 2>& q,
+                      const Segment<T, 2>& s,
+                      double edge_tol = 1e-8)
+{
+  return detail::linear_winding_number(q, s[0], s[1], edge_tol);
+}
+
+/*
+ * \brief Compute the winding number with respect to a 2D triangle
+ *
+ * \param [in] q The query point to test
+ * \param [in] tri The triangle
+ * \param [in] edge_tol The tolerance at which a point is on the line
+ *
+ * \return int The integer winding number
+ */
+template <typename T>
+int winding_number(const Point<T, 2>& q,
+                   const Triangle<T, 2>& tri,
+                   bool useStrictInclusion = false,
+                   double edge_tol = 1e-8)
+{
+  return winding_number(
+    q,
+    Polygon<T, 2>(axom::Array<Point<T, 2>>({tri[0], tri[1], tri[2]})),
+    useStrictInclusion,
+    edge_tol);
+}
+
+/*!
+ * \brief Computes the winding number for a point and a polygon
+ *
+ * \param [in] R The query point to test
+ * \param [in] P The Polygon object to test for containment
+ * \param [in] useStrictInclusion If true, points on the boundary are considered exterior.
+ * \param [in] EPS The tolerance level for collinearity
+ * 
+ * Uses an adapted ray-casting approach that counts quarter-rotation
+ * of vertices around the query point. Current policy is to return 1 on edges
+ * without strict inclusion, 0 on edges with strict inclusion.
+ *
+ * Directly uses algorithm in 
+ * Kai Hormann, Alexander Agathos, "The point in polygon problem for arbitrary polygons"
+ * Computational Geometry, Volume 20, Issue 3, 2001,
+ * 
+ * \return The integer winding number
+ */
+template <typename T>
+int winding_number(const Point<T, 2>& R,
+                   const Polygon<T, 2>& P,
+                   bool useStrictInclusion = false,
+                   double EPS = 1e-8)
+{
+  const int nverts = P.numVertices();
+
+  // If the query is a vertex, return a value interpreted
+  //  as "inside" by evenodd or nonzero protocols
+  if(axom::utilities::isNearlyEqual(P[0][0], R[0], EPS) &&
+     axom::utilities::isNearlyEqual(P[0][1], R[1], EPS))
+    return !useStrictInclusion;
+
+  int winding_num = 0;
+  for(int i = 0; i < nverts; i++)
+  {
+    int j = (i == nverts - 1) ? 0 : i + 1;
+
+    if(axom::utilities::isNearlyEqual(P[j][1], R[1], EPS))
+    {
+      if(axom::utilities::isNearlyEqual(P[j][0], R[0], EPS))
+        return !useStrictInclusion;  // On vertex
+      else if(P[i][1] == R[1] && ((P[j][0] > R[0]) == (P[i][0] < R[0])))
+        return !useStrictInclusion;  // On horizontal edge
+    }
+
+    // Check if edge crosses horizontal line
+    if((P[i][1] < R[1]) != (P[j][1] < R[1]))
+    {
+      double det;
```

Comment:
It would be cleaner and less confusing to read if det was declared in the 2 scopes where it's used.